### PR TITLE
bilibili: init at 1.9.2-1

### PR DESCRIPTION
**Pull request by Jedsek**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/bilibili/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    makeWrapper ${electron}/bin/electron $out/bin/bilibili \
+      --argv0 "bilibili" \
+      --add-flags "$out/opt/app.asar"
+  '';
+
+  meta = with lib; {
+    description = "bilibili client in linux, based on electron";
+    homepage = "https://github.com/msojocs/bilibili-linux";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jedsek ];
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:38]:
```suggestion
    maintainers = with maintainers; [ jedsek ];
    platforms = platforms.unix;
```

File: pkgs/applications/video/bilibili/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ stdenv
+, fetchurl
+, electron
+, lib
+, makeWrapper
+, ...
+} @ args:
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:7]:
```suggestion
, makeWrapper
}:
```

File: pkgs/applications/video/bilibili/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "bilibili";
+  version = "1.9.2-1";
+  src = fetchurl {
+    url = "https://github.com/msojocs/bilibili-linux/releases/download/v1.9.2-1/io.github.msojocs.bilibili_1.9.2-1_amd64.deb";
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:13]:
```suggestion
    url = "https://github.com/msojocs/bilibili-linux/releases/download/v${version}/io.github.msojocs.bilibili_${version}_amd64.deb";
```

File: pkgs/applications/video/bilibili/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    sha256 = "y3dUBImvcIG89m82RaIOa0cxJXIAIGa+n3FJkASacaY=";
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:14]:
```suggestion
    hash = "sha256-y3dUBImvcIG89m82RaIOa0cxJXIAIGa+n3FJkASacaY=";
```

File: pkgs/applications/video/bilibili/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  unpackPhase = ''
+    ar x ${src}
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:18]:
```suggestion
    ar x $src
```

**Comment by wegank** [comment]
@Jedsek Why did you close the PR?

**Comment by Jedsek** [comment]
Oh,sorry,just a small git mistake,wait me a few minutes.

**Comment by wegank** [comment]
I just saw that you're working on the master branch, which is not recommended by [the manual](https://nixos.org/manual/nixpkgs/stable/#chap-submitting-changes). I would suggest creating a branch for each PR, i.e.

```bash
git checkout -b bilibili-init
```

so that you can work on each of them without conflicts.

**Comment by Jedsek** [comment]
Thanks for your suggestions, u are so patient! I will do it :)

File: pkgs/applications/video/bilibili/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    url = "https://github.com/msojocs/bilibili-linux/releases/download/v${version}/io.github.msojocs.bilibili_${version}_amd64.deb";
+    sha256 = "sha256-y3dUBImvcIG89m82RaIOa0cxJXIAIGa+n3FJkASacaY=";
+  };
+
+  unpackPhase = ''
+    ar x $src
+    tar xf data.tar.xz
+  '';
+
+  buildInputs = [ makeWrapper ];
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:21]:
```suggestion
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/applications/video/bilibili/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  unpackPhase = ''
+    runHook preUnpack
+
+    ar x $src
+    tar xf data.tar.xz
+
+    runHook postUnpack
+  '';
+  
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:25]:
```suggestion
  '';
 
  nativeBuildInputs = [ makeWrapper ];
```

File: pkgs/applications/video/bilibili/default.nix [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  '';
+
+  meta = with lib; {
+    description = "bilibili client in linux, based on electron";
+    homepage = "https://github.com/msojocs/bilibili-linux";
+    license = licenses.mit;
+    maintainers = with maintainers; [ jedsek ];
+    platforms = platforms.unix;
+  };  
+}
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:48]:
```suggestion
    platforms = platforms.unix;
  };
}
```

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -6985,6 +6985,12 @@
     githubId = 17029738;
     name = "Jean-Charles Quillet";
   };
+  jedsek = {
+    email = "jedsek@qq.com";
+    github = "jedsek";
+    githubId = 63626406;
+    name = "Jedsek";
+  };
```
Comment (wegank) [inline maintainers/maintainer-list.nix:6993]: This should be moved to a separate commit before `bilibili: init at 1.9.2-1`.

File: pkgs/applications/video/bilibili/default.nix [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+  unpackPhase = ''
+    runHook preUnpack
+
+    ar x $src
+    tar xf data.tar.xz
+
+    runHook postUnpack
+  '';
+ 
+  nativeBuildInputs = [ makeWrapper ];
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:25]:
```suggestion
  '';

  nativeBuildInputs = [ makeWrapper ];
```

Comment (Jedsek) [reply pkgs/applications/video/bilibili/default.nix:25]: there are no whitespace displayed in my editor, so strange

**Comment by wegank** [comment]
`maintainers: add jedsek` should be moved before `bilibili: init at 1.9.2-1`. This can be done in

```
$ git rebase -i HEAD~2
```

Otherwise, LGTM.

File: pkgs/applications/video/bilibili/default.nix [hunk 11]
```diff
@@ -0,0 +1,48 @@
+{ lib
+, stdenv
+, fetchurl
+, electron
+, makeWrapper
+}:
+
+stdenv.mkDerivation rec {
```
Comment (wegank) [inline pkgs/applications/video/bilibili/default.nix:8]: `stdenvNoCC` *might* also work, though I'm not sure.

**Review by wegank — APPROVED** [review]

**Comment by Jedsek** [comment]
ttttttttthanks for your review!
